lib/chef/provider/git.rb

```diff
@@ -72,20 +78,16 @@ def define_resource_requirements
       end
 
       def action_checkout
-        if target_dir_non_existent_or_empty?
-          clone
-          if new_resource.enable_checkout
-            checkout
-          end
-          enable_submodules
-          add_remotes
-        else
-          logger.trace "#{new_resource} checkout destination #{cwd} already exists or is a non-empty directory"
+        clone
+        if new_resource.enable_checkout
+          checkout
         end
+        enable_submodules
+        add_remotes
       end
 
       def action_export
-        action_checkout
+        run_action :checkout
```

Comment:
this is probably not a good idea, it re-runs define_resource_requirements and everything else, which is difficult to reason about.

Reply:
Shouldn't that be exactly what happens, though? This action is an alias for checking out + removing the `.git/` directory when we're done. It seems we should define all the resource requirements for checking out if that's the thing we're going to be doing. Maybe I'm missing something here?